crates/rpc/src/dto/receipt.rs

```diff
@@ -32,6 +32,10 @@ impl From<&pathfinder_common::receipt::ExecutionStatus> for TxnExecutionStatus {
     }
 }
 
+pub struct TxnExecutionStatusWithRevertReason<'a>(
+    pub &'a pathfinder_common::receipt::ExecutionStatus,
+);
+
```

Comment:
I tried to keep these "hidden"/internal types private by embedding them within the actual public parent type e.g. this would have been inside `CommonReceiptProperties`.

If you do embed it, you also don't need the extra type - can just copy the match across.

Reply:
My only problem with that is that then you can't easily test serialization of that part. We can at least remove the `pub` though, that's completely unjustified.